### PR TITLE
cli: catch usage and command-not-found errors

### PR DESCRIPTION
**Pull request by vrothberg**
- don't print help message for usage errors
 - catch command-not-found errors
 - pod create: restore help flag

### CODE REVIEW

**Comment by vrothberg** [comment]
@rhatdan @TomSweeneyRedHat @baude @mheon @umohnani8 PTAL

File: cmd/podman/commit.go [hunk 1]
```diff
@@ -49,15 +49,20 @@ var (
...
+		Usage:        "Create new image based on the changed container",
+		Description:  commitDescription,
+		Flags:        commitFlags,
+		Action:       commitCmd,
+		ArgsUsage:    "CONTAINER [REPOSITORY[:TAG]]",
+		OnUsageError: usageErrorHandler,
 	}
 )
 
+func usageErrorHandler(context *cli.Context, err error, isSubcommand bool) error {
```
Comment (mheon) [inline cmd/podman/commit.go]: Maybe move this over to common.go?

Comment (vrothberg) [reply cmd/podman/commit.go]: That's where I intended to move it ... eww, thanks!

**Comment by rhatdan** [comment]
Does not work for podman command.

```
podman --foobar
```
Prints help followed by the error.

```
Also looks like we loss one of the `-`
./bin/podman run --foobar
flag provided but not defined: -foobar
See 'podman run --help'.
```


```
This preexists.  But it would be nice to get something for this.
./bin/podman pod run
No help topic for 'run'
```

```
This command is printing help
podman pod create --foobar
```


Not sure if this is new

**Comment by vrothberg** [comment]
@rhatdan, thanks for testing!

> Also looks like we loss one of the `-`
```
./bin/podman run --foobar
flag provided but not defined: -foobar
See 'podman run --help'.
```
This predates this change. Docker doesn't have the problem as it doesn't use `urfave/cli`.

I will have a look at the other issues.

**Comment by vrothberg** [comment]
I missed setting the error handler for the app, which fixed the undefined flag issues.

I'll check what we can do for:
```
./bin/podman pod run
No help topic for 'run'
```

**Comment by vrothberg** [comment]
Things are looking better now:

Command not found handling:
```
# ./bin/podman pod run
Command "run" not found.
See `podman pod --help`.

Command "123" not found.
See `podman --help`.
```

Usage error handling:
```
flag needs an argument: -conmon
See 'podman --help'.

flag needs an argument: -h
See 'podman run --help'.

# ./bin/podman pod create --cgroup-parent
flag needs an argument: -cgroup-parent
See 'podman pod create --help'.
```

**Comment by rhatdan** [comment]
Great work @vrothberg.
LGTM

We have a separate bug in         "github.com/urfave/cli"
where it is reporting the --options as -options.
@rh-atomic-bot r+

**Comment by vrothberg** [comment]
Thanks, @rhatdan!

I think that `urfave/cli` can't do much unless they're moving away from golang's `flag` package. I hope to find time to work on https://github.com/golang/go/issues/26822, which could help as well but only for future versions of go.

**Pull request closed without merging**